extensions/rvadapter/src/test/java/com/google/android/agera/rvadapter/RepositoryAdapterTest.java

```diff
@@ -201,6 +201,36 @@ public void shouldCallRecycleForOnViewRecycled() {
   }
 
   @Test
+  public void shouldCallRecycleForOnRebindViewHolder() {
+    when(repositoryPresenter.getItemCount(ALTERNATIVE_REPOSITORY_ITEM)).thenReturn(1);
+    repositoryAdapter.getItemCount(); //Trigger a refresh
+
+    repositoryAdapter.startObserving();
+    repository.accept(ALTERNATIVE_REPOSITORY_ITEM);
+    runUiThreadTasksIncludingDelayedTasks();
+    repositoryAdapter.stopObserving();
+    repositoryAdapter.onBindViewHolder(viewHolder, 0);
+    repositoryAdapter.onBindViewHolder(viewHolder, 1);
```

Comment:
Yep, this was exactly the issue that we had, and that was exactly the fix that was applied. @rybye and @bernardorufino you had some additional concerns though; care to share?

...alternatively, should we make it clear that two presenters could never use the same layout? (with this change that statement is not true though, and we could allow it?)